checks/container.go

```diff
@@ -139,7 +130,22 @@ func fmtContainers(
 			Started:     ctr.StartedAt,
 			Tags:        tags,
 		})
+	}
+	return containers
+}
 
+// chunkContainers formats and chunks the ctrList into a slice of chunks using a specific number of chunks.
+// len(result) MUST EQUAL chunks.
+func chunkContainers(ctrList []*containers.Container, lastRates map[string]util.ContainerRateMetrics, lastRun time.Time, chunks int) [][]*model.Container {
+	perChunk := (len(ctrList) / chunks) + 1
+	chunked := make([][]*model.Container, chunks)
+	chunk := make([]*model.Container, 0, perChunk)
+
+	containers := fmtContainers(ctrList, lastRates, lastRun)
+
+	i := 0
+	for _, ctr := range containers {
+		chunk = append(chunk, ctr)
 		if len(chunk) == perChunk {
 			chunked[i] = chunk
```

Comment:
why not append to chunked instead of keeping track of `i ` ?

Reply:
I simply copied over the existing logic, but I think `chunked` is already initialized with a list of empty slices: `chunked := make([][]*model.Container, chunks)`, if you use `append` instead of using index it would just append after the empty ones at the end. And to do use pre-allocation is mostly for efficiency reasons.

Reply:
I think creating `chunked` with `make([][]*model.Container, 0, chunks)` will have the same efficiency advantages and allow you to use append.